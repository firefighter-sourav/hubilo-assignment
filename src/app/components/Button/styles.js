import styled from 'styled-components'

import { FlexCenter } from 'app/configs/commonStyles'
import { fonts } from 'app/themes/fontPalette'
export const Wrapper = styled(FlexCenter)`
  padding: 15px 40px;
  width: 20%;
  border-radius: 8px;
  background: rgb(32, 32, 32);
  color: white;
  font-weight: bold;
  cursor: pointer;
  font-family: ${fonts.roboto};
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
`
