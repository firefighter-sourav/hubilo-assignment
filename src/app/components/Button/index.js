import PropTypes from 'prop-types'
import React from 'react'

import * as Styles from './styles'

/**
 * Generic Button to use in all the places through out the website
 * Pass onClick function to handle button click
 */
const Button = (props) => {
  const { children, onClick } = props
  return (
    <Styles.Wrapper data-testid="config-button" onClick={onClick}>
      {children}
    </Styles.Wrapper>
  )
}
Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
}
Button.defaultProps = {}
export default Button
