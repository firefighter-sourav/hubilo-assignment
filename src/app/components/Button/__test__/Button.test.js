import {
  cleanup,
  render as testRender,
  fireEvent,
} from '@testing-library/react'
import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

import Button from '../index.js'

afterEach(cleanup)

it('renders ConfigSelect without crasing', () => {
  const onClick = jest.fn()
  const div = document.createElement('div')
  ReactDOM.render(<Button onClick={onClick}>Submit</Button>, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('matches snapshot', () => {
  const onClick = jest.fn()
  const tree = renderer
    .create(<Button onClick={onClick}>Submit</Button>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('onClick should have been called', () => {
  const onClick = jest.fn()
  const utils = testRender(<Button onClick={onClick}>Submit</Button>)
  expect(onClick).toHaveBeenCalledTimes(0)

  const button = utils.getByTestId('config-button')
  fireEvent.click(button)
  expect(onClick).toHaveBeenCalledTimes(1)
  fireEvent.click(button)
  expect(onClick).toHaveBeenCalledTimes(2)
})
