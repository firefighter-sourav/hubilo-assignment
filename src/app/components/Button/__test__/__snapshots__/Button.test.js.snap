// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM sc-gtsrHT sc-dlnjwi kGFBfj NYOTq cDfNFq"
  data-testid="config-button"
  onClick={[MockFunction]}
>
  Submit
</div>
`;
